// ngx-material-timepicker/esm2015/src/app/material-timepicker/directives/overlay.directive.js
import { Directive, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/ngx-material-timepicker-event.service";
export class OverlayDirective {
    constructor(eventService) {
        this.eventService = eventService;
    }
    onClick(e) {
        if (!this.preventClick) {
            this.eventService.dispatchEvent(e);
        }
        e.preventDefault();
    }
}
OverlayDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: OverlayDirective, deps: [{ token: i1.NgxMaterialTimepickerEventService }], target: i0.ɵɵFactoryTarget.Directive });
OverlayDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "12.2.17", type: OverlayDirective, selector: "[overlay]", inputs: { preventClick: ["overlay", "preventClick"] }, host: { listeners: { "click": "onClick($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: OverlayDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[overlay]'
                }]
        }], ctorParameters: function () { return [{ type: i1.NgxMaterialTimepickerEventService }]; }, propDecorators: { preventClick: [{
                type: Input,
                args: ['overlay']
            }], onClick: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
